src/components/list/index.tsx

```diff
@@ -0,0 +1,21 @@
+import React, { FC, PropsWithChildren } from 'react';
+import { List as ChakraList } from '@chakra-ui/react';
+
+export { ListRootProps, ListItemProps } from '@chakra-ui/react';
+
+const { Root, Item } = ChakraList;
+
+Root.displayName = 'List.Root';
+Item.displayName = 'List.Item';
+
+export type ListProps = PropsWithChildren & {
+  variant?: 'icon-inside' | 'icon-outside';
+  size?: 'sm' | 'md';
```

Comment:
Those two are defined in the recipe as `variants` and should be added to props with the `RecipeVariantProps` over declaring them manually

Reply:
https://www.chakra-ui.com/docs/theming/recipes#typescript